sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -171,6 +173,82 @@ case class InSubqueryExec(
   }
 }
 
+/**
+ * The physical node of exists-subquery. This is for support use exists in join's on condition,
+ * since some join type we can't pushdown exists condition, we plan it here
+ */
+case class ExistsExec(child: Expression,
+                      subQuery: String,
+                      plan: BaseSubqueryExec,
+                      exprId: ExprId,
+                      private var resultBroadcast: Broadcast[Boolean] = null)
```

Comment:
why we need broadcast? Can we follow the physical scalar subquery?

Reply:
> why we need broadcast? Can we follow the physical scalar subquery?

Make it can be used by each partition, reduce return data size during compute or return result